content/articles/data-science:emotion-prediction-with-semi-supervised-learning-using-the-RC-algorithm/index.md

```diff
@@ -0,0 +1,192 @@
+"Classification models use one or more features to predict the class label for a new data point. We can predict that the new data point has the same feature values as those that have come before it." 
+### Model Training
+Lets Train our model:
+```dart
+img_size = 48  
+batch_size = 64datagen_train = ImageDataGenerator(horizontal_flip=True)train_generator = datagen_train.flow_from_directory("PATH OF TRAIN FOLDER",  
+                                                    target_size=(img_size,img_size),  
+                                                    color_mode="grayscale",  
+                                                    batch_size=batch_size,  
+                                                    class_mode='categorical',  
+                                                    shuffle=True)datagen_validation = ImageDataGenerator(horizontal_flip=True)  
+validation_generator = datagen_validation.flow_from_directory("PATH OF TEST FOLDER",  
+                                                    target_size=(img_size,img_size),  
+                                                    color_mode="grayscale",  
+                                                    batch_size=batch_size,  
+                                                    class_mode='categorical',  
+                                                    shuffle=False)
+```
+"Recall is measured as how well a model predicts the class label given only some of the training examples from that class." 
+"A model's generalization error is its average prediction error on unseen examples. This value does not equal its overall average accuracy because an algorithm might incorrectly predict in one direction more than it predicts in another (e.g. a model that predicts negative examples to be positive but rarely predicts positive examples to be negative will have a lower average accuracy than a model that predicts all examples to be negative)." 
+"The precision is how well the model predicts the labels of the data points it has seen before." 
+"F1 score is a weighted average of two metrics: precision and recall (inverse document frequency)." 
+"L2-norm is the square root of the sum of squares. It is used to normalize the sum of squared errors, which are the difference between the model's output (e.g., predicted class label) and its training example."
+"Euclidean distance is a common measure of similarity in data. Euclidean distance is based on two main parameters: a set of vectors and an axis perpendicular to that set. Given data points (x1, x2...xn), we can consider the set of all pairs of them (x1, x2,...xn). The Euclidean distance between them (d(x1, x2)) is the length of the line segment connecting them. The cosine similarity is a common shorthand version: 1 - d(x1, x2)/d(x1, x3) * d(x2, x3)."
```

Comment:
This section is improperly formatted. Ensure it is clear and precise. If you are listing points, use a bulleted list.

Reply:
noted